PublishParasoftTestResults/publishtestresults.ts

```diff
@@ -117,6 +123,19 @@ function transformReports(inputReportFiles: string[], index: number)
             } else if (node.name == 'testsuites') {
                 tl.debug("Recognized XUnit report: " + report)
                 reportType = ReportType.XML_XUNIT;
+
+            } else if (node.name == 'CodingStandards' && !bStaticAnalysisResult) {
+                bStaticAnalysisResult = true;
+
+            } else if (node.name == 'Rule') {
+                if (!bLegacyReport) {
+                    ruleAnalyzerPairs.set(node.attributes.id, node.attributes.analyzer);
+                } else if (node.attributes.cat == 'GLOBAL') {
+                    ruleIdsForWhoseCatIsGlobal.push(node.attributes.id);
+                }
+
+            } else if (node.name.endsWith('Viol') && bLegacyReport && bStaticAnalysisResult) {
```

Comment:
[Coding style]: Change the conditions order to be: 
bStaticAnalysisResult && bLegacyReport && node.name.endsWith('Viol')
So the two boolean flag will block a lot of node name check as more efficient.